### PR TITLE
CPLP-1797 daps endpoint

### PR DESCRIPTION
**Pull request by Phil91**
In this PR [CPLP-1797](https://jira.catena-x.net/browse/CPLP-1797) is resolved.

Please also review the database [PR](https://github.com/tractusx-team-portal-onboarding/devenv/pull/179)

### CODE REVIEW

**Comment by jjeroch** [comment]
@Phil91 can we do a minor enhancement? see latest comment in https://jira.catena-x.net/browse/CPLP-1797

**Comment by Phil91** [comment]
> @Phil91 can we do a minor enhancement? see latest comment in https://jira.catena-x.net/browse/CPLP-1797

@jjeroch done - see this [commit](https://github.com/catenax-ng/product-portal-backend/pull/342/commits/74c6b65855b2317b789c82eca89931946edf79a8)

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.Migrations/Migrations/20221122171647_CPLP-1797-AddDapsRegistration.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using Microsoft.EntityFrameworkCore.Migrations;
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20221122171647_CPLP-1797-AddDapsRegistration.cs]: done

File: src/portalbackend/PortalBackend.PortalEntities/Entities/Connector.cs [hunk 2]
```diff
@@ -57,6 +57,8 @@ public Connector(Guid id, string name, string locationId, string connectorUrl)
     [StringLength(2, MinimumLength = 2)]
     public string LocationId { get; set; }
 
+    public bool? DapsRegistrationSuccessful { get; set; }
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/Connector.cs:60]: going to clarify that with @jjeroch

File: src/administration/Administration.Service/Controllers/ConnectorsController.cs [hunk 3]
```diff
@@ -121,12 +140,53 @@ public async Task<CreatedAtRouteResult> CreateConnectorAsync([FromBody] Connecto
...
+    /// <returns>View model of the created connector.</returns>
+    /// <remarks>Example: POST: /api/administration/connectors/trigger-daps/5636F9B9-C3DE-4BA5-8027-00D17A2FECFB</remarks>
+    /// <response code="200">Returns true if the daps call was successful, otherwise false.</response>
+    /// <response code="400">Input parameter are invalid.</response>
+    /// <response code="404">Connector was not found.</response>
+    /// <response code="503">Access to SD factory failed with the given status code.</response>
+    [HttpPost]
+    [Route("trigger-daps/{connectorId:guid}")]
+    [Authorize(Roles = "add_connectors")]
+    [ProducesResponseType(typeof(bool), StatusCodes.Status200OK)]
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/administration/Administration.Service/Controllers/ConnectorsController.cs:183]: done

File: src/administration/Administration.Service/BusinessLogic/DapsService.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+        try
+        {
+            var response = await _httpClient.PostAsync(_settings.DapsUrl, multiPartStream, cancellationToken)
+                .ConfigureAwait(false);
+            return response.IsSuccessStatusCode;
+        }
+        catch (Exception)
+        {
+            return false;
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/administration/Administration.Service/BusinessLogic/DapsService.cs]: changed the behaviour. for the CreateConnector case the process must not fail, therefor we just return null. for the trigger daps case we'll throw an serviceException

File: src/administration/Administration.Service/BusinessLogic/DapsService.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        multiPartStream.Add(new StringContent(clientName), "clientName");
+        multiPartStream.Add(new StringContent(BaseSecurityProfile), "securityProfile");
+        multiPartStream.Add(new StringContent(referringConnector), "referringConnector");
+        multiPartStream.Add(new StringContent(businessPartnerNumber), "regististrationConnector");
+
+        try
+        {
+            var response = await _httpClient.PostAsync(_settings.DapsUrl, multiPartStream, cancellationToken)
+                .ConfigureAwait(false);
+            return response.IsSuccessStatusCode;
```
[the remark this answers is not in the document]
Comment (Phil91) [reply src/administration/Administration.Service/BusinessLogic/DapsService.cs]: safe as above

File: src/portalbackend/PortalBackend.PortalEntities/Entities/Connector.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Phil91) [reply src/portalbackend/PortalBackend.PortalEntities/Entities/Connector.cs:60]: The bool should stay nullable for later processes

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**